server/ServerApp.cpp

```diff
@@ -550,13 +550,21 @@ void ServerApp::NewMPGameInit(const MultiplayerLobbyData& multiplayer_lobby_data
                 const PlayerConnectionPtr player_connection = *established_player_it;
                 if (player_connection->PlayerID() == player_id)
                 {
-                    player_id_setup_data[player_id] = psd;
+                    PlayerSetupData new_psd = psd;
+                    new_psd.m_player_id = player_id;
+                    psds.emplace_back(std::move(new_psd));
                     found_matched_id_connection = true;
                     break;
                 }
             }
-            if (!found_matched_id_connection)
-                ErrorLogger() << "ServerApp::NewMPGameInit couldn't find player setup data for human player with id: " << player_id;
+            if (!found_matched_id_connection) {
+                if (player_id != Networking::INVALID_PLAYER_ID) {
+                    ErrorLogger() << "ServerApp::NewMPGameInit couldn't find player setup data for human player with id: " << player_id << " player name: " << psd.m_player_name;
+                } else {
+                    // player connect later
```

Comment:
There is no player currently connected for the current setup data. A player may connect later, at which time they may be assigned to this data or the corresponding empire.

Reply:
Fixed.